src/h.js

```diff
@@ -1,22 +1,29 @@
-export function h(tag, data) {
-  var node
-  var stack = []
-  var children = []
+function addChild(children, val) {
+  if (val == null || typeof val === 'boolean') return
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
+  if (Array.isArray(val)) {
+    for (var j = 0; j < val.length; j++) {
+      children.push(val[j])
+    }
   }
+  else {
+    children.push(typeof val === 'number' ? val + '' : val)
+  }
+}
 
-  while (stack.length) {
-    if (Array.isArray((node = stack.pop()))) {
-      for (var i = node.length; i--; ) {
-        stack[stack.length] = node[i]
+export function h(tag, data, values) {
+  var children = []
+
+  if (arguments.length > 2) {
+    if (Array.isArray(values)) {
```

Comment:
This block may be better served as:

```js
var isArr = Array.isArray(values), 
  iter = isArr ? values : arguments,
  len = iter.length,
  i = isArr ? 0 : 2;

for (; i < len; i++) {
  addChild(children, iter[i])
}
```

This is because `for-loops` are always (currently) faster if `i` and `len` are declared early. Looks like this saves bytes too?